veps/sig-network/hotpluggable-nad-ref.md

```diff
@@ -0,0 +1,197 @@
+
+### Adjusting when the RestartRequired condition is added
+Currently, during the Sync cycle, the VM controller in virt-controller checks if a `RestartRequired` condition should be added.
+This in turn calls various sub-logics that check for changes in areas like disk, volume, network, memory etc and decide if that requires a restart. The sub-logic for network
+would be updated to not require the `RestartRequired` condition if just the NAD reference (`networkName`) field has been changed.
+
+### Copying network specs from VM to VMI
+During the same Sync cycle, a logic deals checks if interfaces are being hotplugged or modified
+and copies the interfaces specs from VM to VMI specs accordingly. This would be adjusted to sync changes made to the `networkName` field as well.
+
+### Adding the correct migration conditions
+During the updateStatus cycle, the VMI controller in virt-controller evaluates the automatic migration request. Changes to interfaces and networks can trigger 3 states - immediate/pending/no migration.
+In case of hotplug/unplug of interfaces with supported bindings it requests either immediate migration condition (for SR-IOV interfaces) or pending migration (which is a delayed migration - 
+providing enough time for the `Dynamic Networks Controller`, if present to do an `InPlace` hotplug). We would follow a similar approach for swapping the NAD reference.
+So, this would be updated to request an immediate migration in case the interface connected to the changed network uses bridge binding.
+In case the VMI is live-migratable, as before, the WorkloadUpdateController will initiate a migration. Since the new NAD reference is now present on the VMI specs, the target pod's multus annotation 
+will refer to the new NAD.
+
+### Compatibility with clusters that uses Multus Dynamic networks Controller
+#### Option 1
+In order to support in-place hotplug and hot unplug using the [Dynamic Networks Controller](https://github.com/k8snetworkplumbingwg/multus-dynamic-networks-controller) (DNC) the virt-controller updates the 
+multus annotation of the pod based on the difference between the VMI interface status and the specs. DNC then calculates the diff between the new and old annotation, attaches any new network and then removes deleted networks. 
+So trying to change the annotation from `'[{"name":"nad-with-vlan10","namespace":"default","mac":"02:87:f5:51:4c:5d","interface":"poda1363d52898"}]'` to `'[{"name":"nad-with-vlan20","namespace":"default","mac":"02:87:f5:51:4c:5d","interface":"poda1363d52898"}]'` will not work, as DNC will try to attach `nad-with-vlan20`
+without removing `nad-with-vlan10` causing a conflict in mac and interface id and thus failing the attachment.
+
+So, the reconciliation logic in virt-controller will be adjusted to not patch the annotation when only the NAD reference of an existing interface is updated, 
+in order to not break the existing hot-plug / hot-unplug flows.
```

Comment:
So let me see I understand this correctly - in order to successfully change the NAD reference, the MAC address will have to be changed in addition to the NAD? But doesn't this contradict one of the main reasons for this feature?

Reply:
OK, it's a bit clearer now.
Thank you.